sdk/python/kfp/compiler/compiler.py

```diff
@@ -707,14 +625,10 @@ def _get_dependencies(
                     task2=task,
                 )
 
-                # If a task depends on a condition group or a loop group, it
-                # must explicitly dependent on a task inside the group. This
-                # should not be allowed, because it leads to ambiguous
-                # expectations for runtime behaviors.
+                # a task cannot depend on a task created in a for loop group since individual PipelineTask variables are reassigned after each loop iteration
                 dependent_group = group_name_to_group.get(
                     upstream_groups[0], None)
-                if isinstance(dependent_group,
-                              (tasks_group.Condition, tasks_group.ParallelFor)):
+                if isinstance(dependent_group, tasks_group.ParallelFor):
```

Comment:
Shouldn't we add ExitHandler to the check instead of removing Condition from it?

Reply:
Yes, I was thinking about this the wrong way

Reply:
Added in [1c5a898](https://github.com/kubeflow/pipelines/pull/8088/commits/1c5a898dc38e439dbe3ab2b48d59fe433c707135).

Also added support for nested exit handlers to this PR with [ebee976](https://github.com/kubeflow/pipelines/pull/8088/commits/ebee976befe78c8650312625d2ff884ec0397193) and [ba5340c](https://github.com/kubeflow/pipelines/pull/8088/commits/ba5340cfe3d59cfb372d033781fad22be75ea360)